silence-removal/vad.py

```diff
@@ -0,0 +1,172 @@
+        self.__channels = self.get_audio_channels(data = data)
+        if self.__channels == 1:
+            logging.info("This is mono audio")
+            # Reshape data so that it works with np.append(axis = 0)
+            # and other functions that only work with 2d arrays
+            data = data.reshape(len(data), 1)
+            self.__buffer = np.zeros(shape = (0, 1), dtype = np.int16)
+            self.__out_buffer = np.zeros(shape = (0, 1), dtype = np.int16)
+            
+        elif self.__channels == 2:
+            logging.info("This is stereo audio")
+            
+        if self.add_samples(data):
+            while len(self.__buffer) >= self.__buffer_size:
+                # Framing
+                window = self.get_frame()
+                # Add window if threshold is met
+                if self.vad(_frame = window, threshold = 0.01): # speech frame
+                    self.__out_buffer = np.append(self.__out_buffer, window, axis = 0)
+
+    def get_voice_samples(self):
+        # If mono audio, undo the earlier reshaping
+        if self.__channels == 1:
+            self.__out_buffer = self.__out_buffer.reshape(
+                len(self.__out_buffer)
+                )
+            # Check that the resulting shape is 1d instead of 2d
+            logging.debug(self.__out_buffer.shape)
+        return self.__out_buffer
+    
+    def get_audio_channels(self, data):
+        """
+        Function that takes audio data and returns the number of audio   channels as an integer.
+        
+        Arguments:
+        data | numpy.ndarray
+        Audio data from
+        `scipy.io.wavfile.read(filename = 'your_file_name')[1]`.
+        
+        Returns:
+        int
+        1 for mono audio, 2 for stereo audio
+        """
+        # Mono audio is just an array
+        if len(data.shape) == 1:
+            return 1
+        # However stereo audio is an array of arrays where each inner array
+        # is the audio of both sides at that particular time point
+        elif data.shape[1] == 2:
+            return 2
+        else:
+            raise Exception("Error, data is neither mono or stereo!")
+
+# Usage:
+# Convert the `.aac` audio file into `.wav`
+input_filename = sys.argv[1]
+audio_file_extension = re.search(pattern = "\.\w+$",
+                                 string = input_filename).group(0)
+logging.info(f"Audio file extension: {audio_file_extension}")
+if audio_file_extension != ".wav":
+    newFilename = convert_to_wav(input_filename)
+else:
+    newFilename = input_filename
+wav = wf.read(newFilename)
+sample_rate = wav[0]
+audio_data = wav[1]
+logging.info(f"Audio data: {audio_data}")
+
+vad = VoiceActivityDetection()
+# Remove the silences
+vad.process(audio_data)
+# Get the processed audio
+voice_samples = vad.get_voice_samples()
+output_name = sys.argv[2]
+# Write the processed audio into a new `.wav` file
+wf.write(f"{output_name}.wav", sample_rate, voice_samples)
```

Comment:
add endline

Reply:
I've added endlines to all three files. It doesn't show up in the blob, but if you download the raw file there'll be an endline.